shotover/src/hot_reload/protocol.rs

```diff
@@ -23,3 +23,14 @@ pub enum Response {
     //ShutdownOriginalNode,
     Error(String),
 }
+/// Request sent from hot reload server to TcpCodecListener
+pub struct HotReloadListenerRequest {
+    pub return_chan: tokio::sync::oneshot::Sender<HotReloadListenerResponse>,
+}
+
+/// Response sent from TcpCodecListener back to hot reload server
+#[derive(Debug)]
+pub struct HotReloadListenerResponse {
```

Comment:
Currently we set the fd to -1 to indicate an error, but its better to have an explicit error condition rather than magic integer values.
So lets expand this type to support reporting errors.
I am thinking we should turn it into an enum:
```rust
#[derive(Debug)]
pub enum HotReloadListenerResponse {
   HotReloadResponse { port: u16, listener_socket_fd: FileDescriptor },
   NoListenerAvailable,
}
```